wire-swift-generator/src/main/java/com/squareup/wire/swift/SwiftGenerator.kt

```diff
@@ -610,7 +616,11 @@ class SwiftGenerator private constructor(
                 CodeBlock.of("%N", field.name)
               }
             }
-            addStatement("self.%N = %L", field.name, initializer)
+            if (field.default != null) {
+              addStatement("self._%1N = Defaulted(wrappedValue: %2L, defaultValue: %3T.Defaults.%1N)", field.name, initializer, type.typeName)
```

Comment:
General comment mentioned offline:

This would be clearer / easier to read if you did something like:

```kt
// Setup all of our @Defaulted values
addStatement("")
type.fields.filter { it.default != nil }.forEach { field ->
    addStatement("self._%1N = Defaulted(defaultValue: %2T.Defaults.%1N)", field.name, type.typeName)
}

// Populate fields
addStatement("")
type.fields.forEach { /* existing forEach bodies that set `self.foo = initializer` */ }
```

Given that there's N init methods, you could make the codgen for `self._%1N = Defaulted(` be a method that the various places call into.